multicluster/controllers/multicluster/serviceexport_controller.go

```diff
@@ -45,18 +142,470 @@ type ServiceExportReconciler struct {
+			req.String(), "resourceexport", epExportNSName)
+		err := r.endpointsHandler(ctx, req, ep, epResExportName, re, remoteCluster)
+		if err != nil {
+			klog.ErrorS(err, "fail to handle Endpoints change", "endpoints", req.String())
+			return ctrl.Result{}, err
+		}
+	}
 	return ctrl.Result{}, nil
 }
 
+func (r *ServiceExportReconciler) handleServiceDeleteEvent(ctx context.Context, req ctrl.Request, remoteCluster core.RemoteCommonArea) error {
+	svcResExportName := getResourceExportName(localClusterID, req, "service")
+	svcResExport := &mcsv1alpha1.ResourceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      svcResExportName,
+			Namespace: leaderNamespace,
+		},
+	}
+
+	// clean up Service kind of ResourceExport in remote leader cluster
+	svcResExportNamespaced := common.NamespacedName(leaderNamespace, svcResExportName)
+	err := remoteCluster.Delete(ctx, svcResExport, &client.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		klog.ErrorS(err, "fail to delete ResourceExport in remote cluster", "resourceexport",
+			svcResExportNamespaced, "clusterID", leaderClusterID)
+		return err
+	}
+	klog.V(2).InfoS("clean up ResourceExport in remote cluster", "resourceexport", svcResExportNamespaced)
+	return nil
+}
+
+func (r *ServiceExportReconciler) handleEndpointDeleteEvent(ctx context.Context, req ctrl.Request, remoteCluster core.RemoteCommonArea) error {
+	epResExportName := getResourceExportName(localClusterID, req, "endpoints")
+	epResExport := &mcsv1alpha1.ResourceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      epResExportName,
+			Namespace: leaderNamespace,
+		},
+	}
+
+	// clean up Endpoints kind of ResourceExport in remote leader cluster
+	epResExportNamespaced := common.NamespacedName(leaderNamespace, epResExportName)
+	err := remoteCluster.Delete(ctx, epResExport, &client.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		klog.ErrorS(err, "fail to delete ResourceExport in remote cluster", "resourceexport",
+			epResExportNamespaced, "clusterID", leaderClusterID)
+		return err
+	}
+
+	klog.V(2).InfoS("clean up ResourceExport in remote cluster", "resourceexport", epResExportNamespaced)
+	return nil
+}
+
+func (r *ServiceExportReconciler) updateSvcExportStatus(ctx context.Context, req ctrl.Request, cause string) error {
+	svcExport := &k8smcsv1alpha1.ServiceExport{}
+	err := r.Client.Get(ctx, req.NamespacedName, svcExport)
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+	now := metav1.Now()
+	var reason, message *string
+	switch cause {
+	case "not-found":
+		reason = getPointer("not_found_service")
+		message = getPointer("corresponding Service does not exist")
+	case "imported-service":
+		reason = getPointer("imported_service")
+		message = getPointer("corresponding Service is imported service, it's not allowed to export")
+	default:
+		reason = getPointer("invalid_service")
+		message = getPointer("it's not allowed to export")
```

Comment:
Just say "the Service is not valid to export"?

Reply:
done